docs/search_algorithm_switching.md

```diff
@@ -1,37 +1,533 @@
 # Search Algorithm Switching
 
-As part of the Search and Race project work, we've decided to explore allowing users to
-submit their searches using different search algorithms.
+As part of the Search and Race project work, we've decided to explore allowing users to execute their catalog search queries using alternative algorithms used for querying Apache Solr as well as for ranking the search results.
 
-## Context
+## Technical Implementation
 
-We could do this primarily in ruby code or primarily in Solr.
+### Apache Solr Request Handler
 
-### Solr Request Handler
+_(Please refer to [the documentation](https://solr.apache.org/guide/solr/latest/configuration-guide/requesthandlers-searchcomponents.html#RequestHandlersandSearchComponentsinSolrConfig-RequestHandlers) for this component of Apache Solr.)_
 
-Define weights in the solrconfig. Use a sew SearchBuilder to submit searches with "qt=different_algorithm" instead of "qt=search".
-* Pros:
-    * Maximum functionality
-    * hides a bunch of parameters.
-* Cons:
-    * Have to maintain two request handlers, have to deploy.
+In order to provide support at the level of the SolrCloud deployment for the
+Princeton University Library, one must please first focus upon proposing changes
+for the [PUL Solr](https://github.com/pulibrary/pul_solr) GitHub repository.
 
-### Blacklight Request Parameters
+An example of this may be found within the
+[solrconfig.xml](https://github.com/pulibrary/pul_solr/blob/main/solr_configs/catalog-production-v2/conf/solrconfig.xml)
+for the SolrCloud Collection. In this case, the `requestHandler`
+`engineering_search` was added, however, for the purposes of this document, the
+example name `alternative-search` shall be used:
 
-In ruby code, add weights to the search query Blacklight sends. This was done in an [earlier test](https://docs.google.com/document/d/1ROvFH-dy2pWneRmouYZiHvnSBSxGkLt7ygb6CCsx1-Y/edit). Use a new SearchBuilder to submit searches with the required boost params
+```xml
+  <requestHandler name="alternative-search" class="solr.SearchHandler" initParams="searchParams">
+    <!-- Appends means we add this parameter always but the query sent by the client can also add more of them. -->
+    <lst name="appends">
+      <str name="boost">if(termfreq(text,'engineering'),100,0)</str>
+    </lst>
+  </requestHandler>
+```
 
-### Optional indexing strategy
+The attribute `initParams` refers to a set of initial parameters passed to the
+`requestHandler` constructed within Apache Solr, and this is also found within
+the `solrconfig.xml`. With regards to the syntax of these query configuration
+values, please reference [the documentation describing the Solr standard query parser](https://solr.apache.org/guide/solr/latest/query-guide/standard-query-parser.html).
```

Comment:
Can we move 'the' in front of the href text? : '...reference the [documentation describing the Solr standard query parser]'